semeio/reporter/reporter.py

```diff
@@ -0,0 +1,29 @@
+from .serialize_numpy import serialize_numpy
+from .serialize_json import serialize_json
+
+
+class Reporter(object):
+    _serializers = [serialize_numpy, serialize_json]
+
+    def __init__(self):
+        pass
+
+    def report(self, key, val):
```

Comment:
Are we we don't want the user to be explicit about the encoding they want? A single string is valid json for instance. As mentioned in #86 I think a `report(self, val:str)` which takes a report statement and aggregates it and a `publish(self, key:str, val:jsonifiable-object)` that stores data as json. Later, we can extend by `publish_matrix(self, key:str, matrix)` and so forth.

Reply:
I see little value in making more public-facing functions when we can easily detect and handle the type ourselves in an unambiguous way. You are correct in that a `str` is JSON-serializable and produces a JSON, but I don't see how that is a problem.

#86 uses reporting of string data in lieu of logging, which I thought we agreed was better left to Python's `logging` module.